bin/si-smoketest/sdf_service.ts

```diff
@@ -0,0 +1,55 @@
+// sdf_service.ts
+import { SdfApiClient } from "./sdf_api_client.ts";
+
+export class SdfService {
+  private sdfClient: SdfApiClient;
+
+  constructor(sdfClient: SdfApiClient) {
+    this.sdfClient = sdfClient;
+  }
+
+  async listOpenChangeSets() {
+    const response = await this.sdfClient.fetch("/change_set/list_open_change_sets");
+    return await response.json();
+  }
+
+  async createChangeSet(changeSetName: string) {
+    const response = await this.sdfClient.fetch("/change_set/create_change_set", {
+      method: "POST",
+      body: { changeSetName },
+    });
+    return await response.json();
+  }
+
+  async listSchemaVariants(changeSetId: string) {
+    const response = await this.sdfClient.fetch(`/v2/workspaces/${this.sdfClient.workspaceId}/change-sets/${changeSetId}/schema-variants`);
+    return await response.json();
+  }
+
+  async createComponent(payload: Record<string, unknown>) {
+    const response = await this.sdfClient.fetch("/diagram/create_component", {
+      method: "POST",
+      body: payload,
+    });
+    return await response.json();
+  }
+
+  async getDiagram(changeSetId: string) {
+    const response = await this.sdfClient.fetch(`/diagram/get_diagram?visibility_change_set_pk=${changeSetId}&workspaceId=${this.sdfClient.workspaceId}`);
+    return await response.json();
+  }
+
+  async deleteComponents(payload: Record<string, unknown>) {
+    await this.sdfClient.fetch("/diagram/delete_components", {
+      method: "POST",
+      body: payload,
+    });
+  }
+
+  async abandonChangeSet(changeSetId: string) {
+    await this.sdfClient.fetch("/change_set/abandon_change_set", {
+      method: "POST",
+      body: { changeSetId },
+    });
+  }
```

Comment:
nice helpers! I think these should be part of the client itself though - don't see any advantage to separating this logic in two files

Reply:
Sure! I can do that if you would like, I was just trying to avoid creating a megafile with everything under the sun in it

Reply:
Way I see it's better to have a megafile (for a moment) than having to think about what goes in the "client" vs the "service" haha

I'd appreciate the change, yeah

Reply:
Sure, on it!

Reply:
Done!

Reply:
thanks!